sdk/freestanding-execution-engine/src/main.rs

```diff
@@ -437,22 +386,18 @@ fn load_data_sources(cmdline: &CommandLineOptions) -> Vec<DataSourceMetadata> {
         file.read_to_end(&mut buffer).unwrap_or_else(|err| {
             error!(
                 "Could not read data source '{}'.  Error '{}' returned.",
-                fname, err
+                file_path, err
             );
             exit(-1)
         });
 
         // XXX: may panic! if u64 and usize have differing bitwidths...
         let id = u64::try_from(id).unwrap();
+        let file_name = format!("input-{}",id);
+        vfs.lock().unwrap().write(&file_name,&buffer).unwrap();
```

Comment:
Why can't `vfs` be passed as a `&mut VFS` to this function (and other similar functions)?  That way, any locking needed on the virtual FS is only done once, at the top-level?  Though I still don't understand why the VFS needs wrapping in `Arc<Mutex<...>>`.